engine/class_modules/warlock/sc_warlock_affliction.cpp

```diff
@@ -606,6 +650,19 @@ struct malefic_rapture_t : public affliction_spell_t
         p()->buffs.malefic_wrath->trigger();
         p()->procs.malefic_wrath->occur();
       }
+
+      if (p()->sets->has_set_bonus( WARLOCK_AFFLICTION, T28, B2 ) )
+      {
+        timespan_t dot_extension =  p()->sets->set( WARLOCK_AFFLICTION, T28, B2 )->effectN( 2 ).time_value() * 1000;
+        warlock_td_t* td = p()->get_target_data( target );
+
+        if ( td )
```

Comment:
the pointer returned by get_target_data should always be non-null.

Reply:
Considering you have, quite recently, argued against saving multiple checks per-action as being an unnecessary optimization surely a single sanity check to prevent dereferencing a null pointer is a reasonable addition to the program?

There is existing similar usage for this exact purpose in simc, and it should add negligible overhead.

Reply:
If `get_target_data` returns a nullptr, something is wrong and it'd be better to crash than to silently ignore it.

Reply:
If that is the decision made here, then I will happily do a separate commit to remove these checks in whole from the warlock modules (rather than removing one instance in this commit, and others in a separate commit.)

Reply:
I don't have anything against checking it per se, but would then change it to an assert or throw instead of silently doing nothing.

Reply:
Resolved by the most recent commit. After discussing, we have elected to handle it within this commit and not take the route of an assert or throw.